hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/utils/FederationStateStoreFacade.java

```diff
@@ -376,10 +378,20 @@ public void updateApplicationHomeSubCluster(
    */
   public SubClusterId getApplicationHomeSubCluster(ApplicationId appId)
       throws YarnException {
-    GetApplicationHomeSubClusterResponse response =
-        stateStore.getApplicationHomeSubCluster(
+    try {
+      if (isCachingEnabled()) {
+        Map<ApplicationId, SubClusterId> cacheAppSubCluster =
+            (Map<ApplicationId, SubClusterId>)
+                cache.get(buildGetApplicationHomeSubClusterRequest(appId.toString()));
```

Comment:
Extract things a little.
buildGetApplicationHomeSubClusterRequest could also take ApplicationId.

Reply:
I will fix it.

Reply:
Still left.